packages/cli/src/commands/deploy/index.ts

```diff
@@ -460,7 +460,10 @@ export default async (client: Client) => {
 
     if (!localConfig.builds || localConfig.builds.length === 0) {
       // Only add projectSettings for zero config deployments
-      createArgs.projectSettings = { sourceFilesOutsideRootDirectory };
+      createArgs.projectSettings = {
+        ...localConfig.projectSettings,
```

Comment:
this is the line that applies the projectSettings to the deployment

Reply:
You might also need to add this to the `missing_project_settings` error handling.

You can test this by making two deployments to the same project. The first one will return early and enter the `missing_project_settings` branch and ask the use to edit the settings if they wish to override. If the `vercel.json` has settings then those should settings automatically populate instead of asking the user again. Then the second deployment should not hit the `missing_project_settings` branch and accept the deployment on the first api call.

Reply:
It depends if the new project should be created with the settings from vercel.json

Reply:
Now that we have an error above for new projects, I think this should be fine.